packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1512,19 +1513,51 @@ module.exports = (
 
         const { uid, scope } = request.auth.credentials;
         const { ecosystemAnonId } = request.payload;
-        const noneMatchHeader = request.headers['If-None-Match'];
         const scopeSet = ScopeSet.fromArray(scope);
 
+        function parseMultiValueHeader(named) {
+          const value = request.headers[named];
+          if (!value || !value.length) {
+            return [];
+          }
+          return value.toString().split(/,\s?/);
+        }
+
+        function hashAnonId(anonId) {
+          const hash = crypto.createHash('md5');
+          return hash.update(anonId).digest('hex');
+        }
+
+        const ifNoneMatch = parseMultiValueHeader('If-None-Match');
+        const ifMatch = parseMultiValueHeader('If-Match');
```

Comment:
I could see a human error case where new clients unintentionally submit the whole anon_id as the header value. What would you think about throwing if the header is longer than the 32 hex chars needed for an md5 hash?